### DISCUSSION TITLE
Development of a plugin

### DISCUSSION THREAD

**Discussion by preemeijer** [netbox-community/netbox#9472] [Q&A]
Hello all,

I have a question for you all: We have develop an internal tool CMT [1] but came to the conclusion that it's not manageable anymore as a complete custom tool. We decided to look at tools / frameworks that can help us out. 
We eventually have decided that Netbox is the way to go... but... we now want to sponsor the work that needs to be done to make this happen.
Does anybody here want to help of knows somebody that I can talk to to make a new community plugin for netbox that can replace CMT within our company?

Kind regards,
Paul

[1] https://oss.trac.surfsara.nl/cmt

**Comment by ChrisDeh** [comment]
Hi Paul,
we're curently working on setting up a opensource project that is based on a lot of inhouse plugin development that sound quite like the same use case - Happy to get in touch with you!

BR
Christian

**Comment by steffenschumacher** [comment]
We are doing partly the same for config generation - would be happy to align+contribute in a community effort - we’re currently generating day0 configs only, but are starting to work on ansible playbooks for dayN stuff. 
Ideally this could somehow maintain aligned templates for both, so that eg. Logging config template exists once, but can be leveraged for both full day0 and ansible for dayN updates?

**Comment by preemeijer** [comment]
Hello @ChrisDeh and @steffenschumacher,
Thanks for reaction to my question. Today our office is closed. Tomorrow  I will ask my stakeholders and also the people that have made our CMT how to move forward with your reactions. Everybody wants to invest in Netbox because of our current solution is not manageable right now. Hopefully I can make a call very soon?
Kind regards, Paul

**Comment by preemeijer** [comment]
Hello all,
Did not get the opportunity to talk internally who can talk to devs so that there is an alignment on this matter. Hopefully next week I have the chance to talk to the stakeholders of our own CMT.
Kind regards, Paul

**Comment by preemeijer** [comment]
Also if there is a company that can help me do some development for Netbox please let me know.

**Comment by mtinberg** [comment]
You already have a Django-based tool, so porting this to be a Netbox Plugin shouldn't be as much of a stretch as it would be for a totally dissimilar tool, however I'm not sure how much complexity you are going to be able to save by making CMT a plugin, if you are already having trouble managing it, how much will porting it to run under Netbox help save time in the long run, or will porting not affect the parts that are hard to manage?  Is the plan to move config data into Netbox Config Contexts and Custom Fields then use Ansible or Salt to render the config templates and update devices, making only the templates, documentation, import/export/audit scripts custom to your site that need to be maintained?

Getting a few organizations to work together and dedicate part of a staff-members time to work on a shared open-source tool can be a good way to build enough manpower to manage the software project if you don't have enough time to maintain a wholly bespoke system, if you can come to consensus on how it works since each organization is going to have a little different device configuration standards and business rules that the app will need to support, and adding flexibility adds complexity.

I tried to participate in something similar a few years ago with neighboring Universities called NetDash https://github.com/netdash/netdash but aside from Michigan doing the work of creating a framework we didn't have a lot of time to work on it and the project flamed out, although it's always out there and we can pick it back up again when we have the time.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: preemeijer ***@***.***>
Sent: Friday, June 3, 2022 5:26 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Development of a plugin (Discussion #9472)


Hello all,

I have a question for you all: We have develop an internal tool CMT [1] but came to the conclusion that it's not manageable anymore as a complete custom tool. We decided to look at tools / frameworks that can help us out.
We eventually have decided that Netbox is the way to go... but... we now want to sponsor the work that needs to be done to make this happen.
Does anybody here want to help of knows somebody that I can talk to to make a new community plugin for netbox that can replace CMT within our company?

Kind regards,
Paul

[1] https://oss.trac.surfsara.nl/cmt
